include/roboptim/core/function.hh

```diff
@@ -47,6 +47,18 @@
 
 # include <roboptim/core/detail/utility.hh>
 
+// We should probably move these templates out of function.hh
+// Flags used for type checking inside the RobOptim function architecture
+# define ROBOPTIM_IS_DIFFERENTIABLE 1UL << 0
+# define ROBOPTIM_IS_TWICE_DIFFERENTIABLE 1UL << 1
+# define ROBOPTIM_IS_LINEAR 1UL << 2
+# define ROBOPTIM_IS_NUMERIC_LINEAR 1UL << 3
+# define ROBOPTIM_IS_QUADRATIC 1UL << 4
+# define ROBOPTIM_IS_NUMERIC_QUADRATIC 1UL << 5
+# define ROBOPTIM_IS_CONSTANT 1UL << 6
+# define ROBOPTIM_IS_POLYNOMIAL 1UL << 7
```

Comment:
Good point.
Why unsigned long ? Because at first I tried to handle the n-times differentiable functions, stocking the N template parameter in the second (or first, depending on which way you look at it) part of the unsigned long. If we stick to what is available now (no n-times differentiable function typing flags) an int is really enough.
If so, there is indeed no need to keep ugly macros, and the enum seems the way to do it.